controllers/webhooks/networking/security_groups/validator_test.go

```diff
@@ -0,0 +1,285 @@
+			actualContext, _, actualNamespace, actualResource := duplicateValidator.ValidateCreateArgsForCall(0)
+			Expect(actualContext).To(Equal(ctx))
+			Expect(actualNamespace).To(Equal(securityGroup.Namespace))
+			Expect(actualResource).To(Equal(securityGroup))
+			Expect(actualResource.UniqueValidationErrorMessage()).To(Equal("Security group with name '" + securityGroup.Spec.DisplayName + "' already exists."))
+		})
+
+		When("the security group name is a duplicate", func() {
+			BeforeEach(func() {
+				duplicateValidator.ValidateCreateReturns(errors.New("foo"))
+			})
+
+			It("denies the request", func() {
+				Expect(retErr).To(MatchError("foo"))
+			})
+		})
+	})
+
+	Describe("ValidateUpdate", func() {
+		var updatedSecurityGroup *korifiv1alpha1.CFSecurityGroup
+
+		BeforeEach(func() {
+			updatedSecurityGroup = securityGroup.DeepCopy()
+			updatedSecurityGroup.Spec.DisplayName = "the-new-name"
+		})
+
+		JustBeforeEach(func() {
+			_, retErr = validatingWebhook.ValidateUpdate(ctx, securityGroup, updatedSecurityGroup)
+		})
+
+		It("invokes the validator correctly", func() {
+			Expect(duplicateValidator.ValidateUpdateCallCount()).To(Equal(1))
+			actualContext, _, actualNamespace, oldResource, newResource := duplicateValidator.ValidateUpdateArgsForCall(0)
+			Expect(actualContext).To(Equal(ctx))
+			Expect(actualNamespace).To(Equal(securityGroup.Namespace))
+			Expect(oldResource).To(Equal(securityGroup))
+			Expect(newResource).To(Equal(updatedSecurityGroup))
+		})
+
+		When("the security group is being deleted", func() {
+			BeforeEach(func() {
+				updatedSecurityGroup.DeletionTimestamp = &metav1.Time{Time: time.Now()}
+			})
+
+			It("does not return an error", func() {
+				Expect(retErr).NotTo(HaveOccurred())
+			})
+		})
+
+		When("the new security group name is a duplicate", func() {
+			BeforeEach(func() {
+				duplicateValidator.ValidateUpdateReturns(errors.New("foo"))
+			})
+
+			It("denies the request", func() {
+				Expect(retErr).To(MatchError("foo"))
+			})
+		})
+	})
+
+	Describe("ValidateDelete", func() {
+		JustBeforeEach(func() {
+			_, retErr = validatingWebhook.ValidateDelete(ctx, securityGroup)
+		})
+
+		It("allows the request", func() {
+			Expect(retErr).NotTo(HaveOccurred())
+		})
+
+		It("invokes the validator correctly", func() {
+			Expect(duplicateValidator.ValidateDeleteCallCount()).To(Equal(1))
+			actualContext, _, actualNamespace, actualResource := duplicateValidator.ValidateDeleteArgsForCall(0)
+			Expect(actualContext).To(Equal(ctx))
+			Expect(actualNamespace).To(Equal(securityGroup.Namespace))
+			Expect(actualResource).To(Equal(securityGroup))
+		})
+
+		When("delete validation fails", func() {
+			BeforeEach(func() {
+				duplicateValidator.ValidateDeleteReturns(errors.New("foo"))
+			})
+
+			It("disallows the request", func() {
+				Expect(retErr).To(MatchError("foo"))
+			})
+		})
+	})
+
+	Describe("Validating the security group rules", func() {
```

Comment:
We should have a test to verify that validation is performed on `Update` as well. It does not have to repeat the complete validation test, maybe just one specific case in order to verify that invalid updates are rejected